utils/src/ooinstall/openshift_ansible.py

```diff
@@ -175,30 +184,20 @@ def write_proxy_settings(base_inventory):
         pass
 
 
-# pylint: disable=too-many-branches
 def write_host(host, role, inventory, schedulable=None):
     global CFG
 
     if host.preconfigured:
         return
 
     facts = ''
-    if host.ip:
-        facts += ' openshift_ip={}'.format(host.ip)
-    if host.public_ip:
-        facts += ' openshift_public_ip={}'.format(host.public_ip)
-    if host.hostname:
-        facts += ' openshift_hostname={}'.format(host.hostname)
-    if host.public_hostname:
-        facts += ' openshift_public_hostname={}'.format(host.public_hostname)
-    if host.containerized:
-        facts += ' containerized={}'.format(host.containerized)
+    for prop in HOST_VARIABLES_MAP:
```

Comment:
I'm good with deferring improvements here until later...  It sounds like we may need some type of shared base class (or even a python mixin) to handle attribute mapping, but I definitely want to see us more more towards self-managing classes for a lot of this stuff to avoid having spaghetti code. Ideally, we should be able to take a top level deployment object and serialize it either into a config object or to a inventory (or inventory and group_vars/host_vars) easily.